logic/LogicManager.java

```diff
@@ -122,21 +133,21 @@ public void revalidateAndReindexRules() {
         }
 
         // using the new index, validate new rules are stratifiable (eg. do not cause cycles through a negation)
-        validateCyclesThroughNegations(conceptMgr, this);
+        validateCyclesThroughNegations();
```

Comment:
right after clearing `logicCache.rule()`, we should also clear `logicCache.unifier()`